### ISSUE TITLE
Bold text disappears

### ISSUE THREAD

**Issue by raineorshine** [cybersemics/em#1540]
There is a regression that deletes text as soon as it is made bold.

## Steps to Reproduce

1. Create a thought with text `ab`
2. Select `b` and hit <kbd>Ctrl</kbd> + <kbd>b</kbd> to make it bold.
3. Move the caret to the end of the thought
4. Hit Enter

## Current Behavior

- a
- [cursor]

## Expected Behavior

- a**b**
- [cursor]

**Comment by karunkop** [comment]
@raineorshine  Upon investigation I found out that, the issue was due to  `stripEmptyFormattingTags` regex in (https://github.com/cybersemics/em/blob/dev/src/util/stripEmptyFormattingTags.ts) file.  I know how to fix this exact issue by tweaking the regex, however I have some concerns.

The regex expression used in this file  was introduced in PR (https://github.com/cybersemics/em/pull/1515) to solve caret issue. Our goal was to strip empty tags upfront right after pasting/writing it. The regex is still unable to handle if there are multiple `nested` empty tags. So i am thinking that regex might not be the optimal way to solve this issue. I was thinking of using `himalaya parser` to identify tags and get rid of those tags which have empty content.  

Let me know your thought on this approach! Thanks!
